### PR TITLE
ci: remove e2e test bottleneck

### PR DESCRIPTION
**Pull request by ggilmore**
DO NOT MERGE until https://github.com/sourcegraph/infrastructure/pull/1881 is approved. 

The `concurrency: 1` setting can now be removed since each buildkite agent will have its own personal docker-in-docker daemon.

### CODE REVIEW

File: dev/ci/e2e.sh [hunk 1]
```diff
@@ -16,14 +16,9 @@ if curl --output /dev/null --silent --head --fail $URL; then
     exit 1
 fi
 
-# Switch the default Docker host to the dedicated e2e testing host
-export DOCKER_HOST="$E2E_DOCKER_HOST"
-export DOCKER_PASSWORD="$E2E_DOCKER_PASSWORD"
-export DOCKER_USERNAME="$E2E_DOCKER_USERNAME"
```
Comment (keegancsmith) [inline dev/ci/e2e.sh:22]: removing these variables in the other PR mean people will need to rebase once the agent is out. Mayube in the other PR stil set the E2E ones

Comment (felixfbecker) [reply dev/ci/e2e.sh:22]: Didn't we have some neat way to require people to rebase to a certain commit? I can't find it anymore

Comment (keegancsmith) [reply dev/ci/e2e.sh:22]: we do, but its not needed in this case. We should just not unset those variables right now.

Comment (ggilmore) [reply dev/ci/e2e.sh:22]: I will put back those env vars in the other PR, and I'll remove them after a month.

**Pull request merged**